forge-gui/src/main/java/forge/player/HumanPlay.java

```diff
@@ -104,6 +104,8 @@ public final static boolean playSpellAbility(final PlayerControllerHuman control
                     if (rollback.isInZone(ZoneType.Exile)) {
                         rollback.addMayLookTemp(p);
                     }
+                } else {
```

Comment:
else is wrong
just move it into the block above

Reply:
in theory `undoAlternateHost` could handle reverting `castFaceDown` too